### ISSUE TITLE
show full image

### ISSUE THREAD

**Issue by ebgoldstein** [UNCG-DAISY/Coastal-Image-Labeler#79]
(so users don't need to click the `^` on the top right)

**Comment by ShahNafisRafique** [comment]
Yes will do! There is actually one thing I have to say about this. So since I do image compression the first time a image is request (across all images) it takes some time. This time is rather inconsistent ranging from a few seconds to 30 or so seconds. If its the later the user would have to toggle the `^` open and close for the image to re-render. This only happens once, so once image X has been compressed it can be retrieved faster for the next time (limited by the VM network I assume). There might be a better way but pre-compressing images was too costly.

**Comment by ebgoldstein** [comment]
ok, i see this problem... so pre-compression took too long?

**Comment by ShahNafisRafique** [comment]
Yea I left the VM to run for 8 hours and it only compressed around 1 and a half archives. Could be a language issue or maybe I just didnt handle it correctly as I told it to compress one image at a time.

**Comment by ebgoldstein** [comment]
i think we need precompression..

**Comment by ShahNafisRafique** [comment]
any suggestions? I could look into improving the CLI performance wise or maybe there is a better compression tool out there?

**Comment by ebgoldstein** [comment]
let table this RN... and have a think... @somyamohanty , i want to loop you in for this one.

**Comment by ShahNafisRafique** [comment]
I got jpegoptim on the VM and I picked a random test image, its an image of just water, and after running
```
jpegoptim -d /datadrive/compressed -p /datadrive/archives/Florence/20180915a_jpgs/jpgs/C25870216.jpg
```
The output is 
```
jpegoptim: skipping directory: /datadrive/compressed
/datadrive/archives/Florence/20180915a_jpgs/jpgs/C25870216.jpg 10328x7760 24bit N Exif ICC IPTC XMP JFIF  [OK] 3532269 --> 3532269 bytes (0.00%), sk        ipped.
```
Meaning it cant compress without losing quality. So if I do compression with loss in quality (lets say down to 40%)
```
jpegoptim: skipping directory: /datadrive/compressed
/datadrive/archives/Florence/20180915a_jpgs/jpgs/C25870216.jpg 10328x7760 24bit N Exif ICC IPTC XMP JFIF  [OK] 3532269 --> 928183 bytes (73.72%), optimized.
jpegoptim: failed to reset output file group/owner
```
Which results in the file size going from `3.53mb` to `.928mb`. 

So the question is

**How much should the loss in quality be? Should it be set to `m40` which means 60% loss? or 40% loss?**
I didnt notice a difference between the two at 40% but I dont know what I am looking for to see a change in quality

**Comment by ShahNafisRafique** [comment]
For the sake of testing I did `m60` which means 40% loss in quality.

**Comment by ShahNafisRafique** [comment]
@ebgoldstein So after leaving the VM running over night starting around 10 am it ran until 12 midnight. It got through Dorian and Florence and some of Michael. The problem was that

```
mkdir: cannot create directory ‘/datadrive/compressed/archives/Michael/20181013a_jpgs’: No space left on device
mkdir: cannot create directory ‘/datadrive/compressed/archives/Michael/20181013a_jpgs/jpgs’: No such file or directory
``

and then after that it logs
```
jpegoptim: invalid argument for option -d, --dest
```
I am not sure how to check the size of the vm, cause from doing `df -h` it doesnt seem to be out of space. And with `du -hs /datadrive/compressed` says the compressed images are 119gb vs the 839gb of the uncompressed

I can edit the bash script to resume from where it left off. But I am not sure about the disk space

It took `48861` seconds which is `13` hours.

**Comment by ShahNafisRafique** [comment]
Actually running `df` shows that the datadrive is full
`/dev/sdb1      1055839668 1002136244         0 100% /datadrive`

**Comment by ShahNafisRafique** [comment]
It seems all images have been compressed. One weird thing is in the database theres 56.2k images recorded(56180). But if I run
```
find /datadrive2/compressed/archives -type f | wc -l
```
it says there is 52383 images across all 3 catalogs. So I am not sure why theres a difference. In any case I added a fall back for the compressed URL which will show uncompressed if there is no compressed image.

**Comment by ShahNafisRafique** [comment]
After checking the database I see that in the DB 
```
Catalog Michael has 5 archives and 9600 images
```
while the compressed folder has
```
Michael 5802
```
